cloud/vmware/vmware_vswitch.py

```diff
@@ -82,82 +82,101 @@
 
 
 def find_vswitch_by_name(host, vswitch_name):
-    for vss in host.config.network.vswitch:
-        if vss.name == vswitch_name:
-            return vss
-    return None
-
-
-# Source from
-# https://github.com/rreubenur/pyvmomi-community-samples/blob/patch-1/samples/create_vswitch.py
-
-def state_create_vswitch(module):
-
-    switch_name = module.params['switch_name']
-    number_of_ports = module.params['number_of_ports']
-    nic_name = module.params['nic_name']
-    mtu = module.params['mtu']
-    host = module.params['host']
-
-    vss_spec = vim.host.VirtualSwitch.Specification()
-    vss_spec.numPorts = number_of_ports
-    vss_spec.mtu = mtu
-    vss_spec.bridge = vim.host.VirtualSwitch.BondBridge(nicDevice=[nic_name])
-    host.configManager.networkSystem.AddVirtualSwitch(vswitchName=switch_name, spec=vss_spec)
-    module.exit_json(changed=True)
-
-
-def state_exit_unchanged(module):
-    module.exit_json(changed=False)
-
-
-def state_destroy_vswitch(module):
-    vss = module.params['vss']
-    host = module.params['host']
-    config = vim.host.NetworkConfig()
-
-    for portgroup in host.configManager.networkSystem.networkInfo.portgroup:
-        if portgroup.spec.vswitchName == vss.name:
-            portgroup_config = vim.host.PortGroup.Config()
-            portgroup_config.changeOperation = "remove"
-            portgroup_config.spec = vim.host.PortGroup.Specification()
-            portgroup_config.spec.name = portgroup.spec.name
-            portgroup_config.spec.vlanId = portgroup.spec.vlanId
-            portgroup_config.spec.vswitchName = portgroup.spec.vswitchName
-            portgroup_config.spec.policy = vim.host.NetworkPolicy()
-            config.portgroup.append(portgroup_config)
-
-    host.configManager.networkSystem.UpdateNetworkConfig(config, "modify")
-    host.configManager.networkSystem.RemoveVirtualSwitch(vss.name)
-    module.exit_json(changed=True)
-
-
-def state_update_vswitch(module):
-    module.exit_json(changed=False, msg="Currently not implemented.")
-
-
-def check_vswitch_configuration(module):
-    switch_name = module.params['switch_name']
-    content = connect_to_api(module)
-    module.params['content'] = content
-
-    host = get_all_objs(content, [vim.HostSystem])
-    if not host:
-        module.fail_json(msg="Unble to find host")
-
-    host_system = host.keys()[0]
-    module.params['host'] = host_system
-    vss = find_vswitch_by_name(host_system, switch_name)
-
-    if vss is None:
-        return 'absent'
-    else:
-        module.params['vss'] = vss
-        return 'present'
+        for vss in host.config.network.vswitch:
+            if vss.name == vswitch_name:
+                return vss
+        return None
+
+
+class VMwareHostVirtualSwitch(object):
```

Comment:
@chrrrles I do think that these modules could be refactored using a parent and child classes.  I have eight more module updates similar to this one - I don't know about merging all those classes into the single `module_utils\vmware.py` is the approach we want to take.

What my main intention in these PRs was to resolve an issue I unknowingly introduced for version 2.0.  We are also in the process of open sourcing the playbooks to bare-metal deploy vSphere.  That effort is contingent on these modules being updated.

That said I am sure @mtnbikenc and I would be willing to hear any suggestions for making these modules better.